pallets/inactivity-tracking/src/lib.rs

```diff
@@ -304,6 +319,53 @@ pub mod pallet {
             }
         }
 
+        /// Internal function to populate the current session active collator records with collators
+        /// part of inactive chains.
+        pub fn process_inactive_chains_for_session() {
+            match <CurrentActivityTrackingStatus<T>>::get() {
+                ActivityTrackingStatus::Disabled { .. } => return,
+                ActivityTrackingStatus::Enabled { start, end: _ } => {
+                    if start > T::CurrentSessionIndex::session_index() {
+                        return;
+                    }
+                }
+            }
+            let active_chains = <ActiveContainerChainsForCurrentSession<T>>::get();
+            let _ = <ActiveCollatorsForCurrentSession<T>>::try_mutate(
+                |active_collators| -> DispatchResult {
+                    let container_chains_with_collators =
+                        T::CurrentCollatorsFetcher::container_chains_with_collators(
+                            ForSession::Current,
+                        );
+                    for (para_id, collator_ids) in container_chains_with_collators.iter() {
+                        if !active_chains.contains(para_id)
+                            && !T::ParathreadHelper::is_parathread(para_id)
```

Comment:
```suggestion
                            && !T::ParaFilter::is_parathread(para_id)
```
let's be more generic, and call it filter

Reply:
so basically I would make it inject a vec of para ids and return a vec of paraids

Reply:
Naming updated in https://github.com/moondance-labs/tanssi/pull/899/commits/caa56e8e3a9a064a6ad44f0989b3b7423369eb6f

Reply:
ParaFilter refactored to return a vector of parathreads in https://github.com/moondance-labs/tanssi/pull/899/commits/14c88e43259860951ca9797cc60e1e845dbd18a2